mods/content/pixel_shifting/pixel_shifting.dm

```diff
@@ -0,0 +1,44 @@
+/mob
+	/// Wether the mob is currently pixel shifted or not
+	var/is_shifted
+	/// If the mob is currently pixel shifting
+	var/shifting
+
+/mob/proc/unpixel_shift()
+	return
+
+/mob/living/unpixel_shift()
+	if(is_shifted)
+		is_shifted = FALSE
+		pixel_x = initial(pixel_x)
+		pixel_y = initial(pixel_y)
+
+/mob/proc/pixel_shift(direction)
+	return
+
+/mob/living/pixel_shift(direction)
+	switch(direction)
+		if(NORTH)
+			if(!canface())
+				return FALSE
+			if(pixel_y <= 16)
+				pixel_y++
+				is_shifted = TRUE
+		if(EAST)
+			if(!canface())
+				return FALSE
+			if(pixel_x <= 16)
+				pixel_x++
+				is_shifted = TRUE
+		if(SOUTH)
+			if(!canface())
+				return FALSE
+			if(pixel_y >= -16)
+				pixel_y--
+				is_shifted = TRUE
+		if(WEST)
+			if(!canface())
+				return FALSE
+			if(pixel_x >= -16)
+				pixel_x--
+				is_shifted = TRUE
```

Comment:
Or you can just copypaste my proc:

```
#define SHIFT_MAX 8

/mob/proc/pixel_shift(dir)
	if(!canface() || !can_shift())
		return FALSE

	switch(dir)
		if(NORTH)
			if(pixel_y <= SHIFT_MAX)
				pixel_y++
		if(EAST)
			if(pixel_x <= SHIFT_MAX)
				pixel_x++
		if(SOUTH)
			if(pixel_y >= -SHIFT_MAX)
				pixel_y--
		if(WEST)
			if(pixel_x >= -SHIFT_MAX)
				pixel_x--
		else
			CRASH("Invalid argument supplied!")

	is_shifted = TRUE
	UPDATE_OO_IF_PRESENT

#undef SHIFT_MAX
```

Note about the `UPDATE_OO_IF_PRESENT`. It will update a openspace overlay above you (so people above will see your character is shifted).